pkg/proxy/winkernel/proxier.go

```diff
@@ -173,15 +196,97 @@ func conjureMac(macPrefix string, ip net.IP) string {
 	return "02-11-22-33-44-55"
 }
 
-func newEndpointInfo(ip string, port uint16, isLocal bool, hns HostNetworkService) *endpointsInfo {
+func (proxier *Proxier) endpointsMapChange(oldEndpointsMap, newEndpointsMap proxy.EndpointsMap) {
+	for svcPortName := range oldEndpointsMap {
+		proxier.onEndpointsMapChange(&svcPortName)
+	}
+
+	for svcPortName := range oldEndpointsMap {
+		proxier.onEndpointsMapChange(&svcPortName)
+	}
+}
+
+func (proxier *Proxier) onEndpointsMapChange(svcPortName *proxy.ServicePortName) {
+
+	svc, exists := proxier.serviceMap[*svcPortName]
```

Comment:
May want to add some nil handling if `*svcPortName` is nil.

Reply:
It is not expected to be nil as it is obtained from enumeration of the map. And  since it is passed to an internal function i skipped any validation.